File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.go
func containsNearbyDuplicate(nums []int, k int) bool {
    seen := make(map[int]int)
    for i, n := range nums {
        if j, ok := seen[n]; ok && abs(i-j) <= k {
            return true
        }
        seen[n] = i
    }
    return false
}

func abs(x int) int {
    if x < 0 {
        return -x
    }
    return x
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        seen = {}
        for i in range(len(nums)):
            j = seen.get(nums[i])
            if j != None and abs(i-j) <= k:
                return True
            seen[nums[i]] = i
        return False    
                